### PR TITLE
MMF-1252 connected mode

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: scripts/prepare.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ if (!fs.existsSync("analyzers")) {
 }
 
 downloadIfNeeded(
-  "https://repox.sonarsource.com/sonarsource/org/sonarsource/sonarlint/core/sonarlint-language-server/3.1.0.1376/sonarlint-language-server-3.1.0.1376.jar",
+  "https://repox.sonarsource.com/sonarsource/org/sonarsource/sonarlint/core/sonarlint-language-server/3.5.0.1548/sonarlint-language-server-3.5.0.1548.jar",
```
Comment (SonarTech) [inline scripts/prepare.js]: Code Smell Code Smell: Replace double quotes by single quotes ([javascript:SingleQuote](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ASingleQuote))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.vscode%3Asonarlint-vscode&pullRequest=15&issues=AWNdSse_nxz_2987348L&open=AWNdSse_nxz_2987348L)

[ISSUE_KEY]: AWNdSse_nxz_2987348L

File: scripts/prepare.js [hunk 2]
```diff
@@ -12,23 +18,23 @@ if (!fs.existsSync("analyzers")) {
...
   "analyzers/sonarphp.jar"
 );
 downloadIfNeeded(
-  "http://repo1.maven.org/maven2/org/sonarsource/python/sonar-python-plugin/1.8.0.1496/sonar-python-plugin-1.8.0.1496.jar",
+  `https://repo1.maven.org/maven2/org/sonarsource/python/sonar-python-plugin/${sonarPythonVersion}/sonar-python-plugin-${sonarPythonVersion}.jar`,
   "analyzers/sonarpython.jar"
 );
 downloadIfNeeded(
-  "https://repox.sonarsource.com/sonarsource/org/sonarsource/typescript/sonar-typescript-plugin/1.5.0.2122/sonar-typescript-plugin-1.5.0.2122.jar",
+  `https://repox.sonarsource.com/sonarsource/org/sonarsource/typescript/sonar-typescript-plugin/${sonarTsVersion}/sonar-typescript-plugin-${sonarTsVersion}.jar`,
```
Comment (SonarTech) [inline scripts/prepare.js]: Code Smell Code Smell: Split this 161 characters long line (which is greater than 120 authorized). ([javascript:LineLength](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ALineLength))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.vscode%3Asonarlint-vscode&pullRequest=15&issues=AWNi0H6X6IBDU_gafCV1&open=AWNi0H6X6IBDU_gafCV1)

[ISSUE_KEY]: AWNi0H6X6IBDU_gafCV1

File: scripts/prepare.js [hunk 3]
```diff
@@ -12,23 +18,23 @@ if (!fs.existsSync("analyzers")) {
 }
 
 downloadIfNeeded(
-  "https://repox.sonarsource.com/sonarsource/org/sonarsource/sonarlint/core/sonarlint-language-server/3.1.0.1376/sonarlint-language-server-3.1.0.1376.jar",
+  `https://repox.sonarsource.com/sonarsource/org/sonarsource/sonarlint/core/sonarlint-language-server/${languageServerVersion}/sonarlint-language-server-${languageServerVersion}.jar`,
   "server/sonarlint-ls.jar"
 );
 downloadIfNeeded(
-  "http://repo1.maven.org/maven2/org/sonarsource/javascript/sonar-javascript-plugin/4.0.0.5862/sonar-javascript-plugin-4.0.0.5862.jar",
+  `https://repo1.maven.org/maven2/org/sonarsource/javascript/sonar-javascript-plugin/${sonarJsVersion}/sonar-javascript-plugin-${sonarJsVersion}.jar`,
```
Comment (SonarTech) [inline scripts/prepare.js]: Code Smell Code Smell: Split this 150 characters long line (which is greater than 120 authorized). ([javascript:LineLength](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ALineLength))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.vscode%3Asonarlint-vscode&pullRequest=15&issues=AWNi0H6X6IBDU_gafCVy&open=AWNi0H6X6IBDU_gafCVy)

[ISSUE_KEY]: AWNi0H6X6IBDU_gafCVy

File: scripts/prepare.js [hunk 4]
```diff
@@ -12,23 +18,23 @@ if (!fs.existsSync("analyzers")) {
...
   "analyzers/sonarjs.jar"
 );
 downloadIfNeeded(
-  "http://repo1.maven.org/maven2/org/sonarsource/php/sonar-php-plugin/2.12.0.2871/sonar-php-plugin-2.12.0.2871.jar",
+  `https://repo1.maven.org/maven2/org/sonarsource/php/sonar-php-plugin/${sonarPhpVersion}/sonar-php-plugin-${sonarPhpVersion}.jar`,
```
Comment (SonarTech) [inline scripts/prepare.js]: Code Smell Code Smell: Split this 131 characters long line (which is greater than 120 authorized). ([javascript:LineLength](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ALineLength))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.vscode%3Asonarlint-vscode&pullRequest=15&issues=AWNi0H6X6IBDU_gafCVz&open=AWNi0H6X6IBDU_gafCVz)

[ISSUE_KEY]: AWNi0H6X6IBDU_gafCVz

File: scripts/prepare.js [hunk 5]
```diff
@@ -12,23 +18,23 @@ if (!fs.existsSync("analyzers")) {
...
   "analyzers/sonarphp.jar"
 );
 downloadIfNeeded(
-  "http://repo1.maven.org/maven2/org/sonarsource/python/sonar-python-plugin/1.8.0.1496/sonar-python-plugin-1.8.0.1496.jar",
+  `https://repo1.maven.org/maven2/org/sonarsource/python/sonar-python-plugin/${sonarPythonVersion}/sonar-python-plugin-${sonarPythonVersion}.jar`,
```
Comment (SonarTech) [inline scripts/prepare.js]: Code Smell Code Smell: Split this 146 characters long line (which is greater than 120 authorized). ([javascript:LineLength](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ALineLength))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.vscode%3Asonarlint-vscode&pullRequest=15&issues=AWNi0H6X6IBDU_gafCV0&open=AWNi0H6X6IBDU_gafCV0)

[ISSUE_KEY]: AWNi0H6X6IBDU_gafCV0

File: scripts/prepare.js [hunk 6]
```diff
@@ -12,23 +18,23 @@ if (!fs.existsSync("analyzers")) {
 }
 
 downloadIfNeeded(
-  "https://repox.sonarsource.com/sonarsource/org/sonarsource/sonarlint/core/sonarlint-language-server/3.1.0.1376/sonarlint-language-server-3.1.0.1376.jar",
+  `https://repox.sonarsource.com/sonarsource/org/sonarsource/sonarlint/core/sonarlint-language-server/${languageServerVersion}/sonarlint-language-server-${languageServerVersion}.jar`,
```
Comment (SonarTech) [inline scripts/prepare.js]: Code Smell Code Smell: Split this 183 characters long line (which is greater than 120 authorized). ([javascript:LineLength](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ALineLength))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.vscode%3Asonarlint-vscode&pullRequest=15&issues=AWNi0H6X6IBDU_gafCVx&open=AWNi0H6X6IBDU_gafCVx)

[ISSUE_KEY]: AWNi0H6X6IBDU_gafCVx

**Comment by darlandemarco** [comment]
Great job! Does anybody knows when this will  be released?

**Comment by janos-ss** [comment]
@darlandemarco A bit more work is needed here. It will be released this week or the next.

**Comment by darlandemarco** [comment]
@janos-ss great work! 👍I'm very anxious for this release, it will help my organization a lot! 

Thank you so much! :)

**Pull request merged**